### PR TITLE
Implemented simpleSQL() method and out() test.

### PR DESCRIPTION
**Pull request by DenysMelnychenko**
connect to #21

### CODE REVIEW

File: src/main/java/juja/sqlcmd/Application.java [hunk 1]
```diff
@@ -1,6 +1,153 @@
 package juja.sqlcmd;
 
+import java.sql.*;
+
 public class Application {
-    public static void main(String[] args) {
+
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: Check the blancklines and whitespaces during the code  (http://bit.ly/2j2qPE1)

File: src/main/java/juja/sqlcmd/Application.java [hunk 2]
```diff
@@ -1,6 +1,153 @@
...
+        try (Statement statement = connection.createStatement()) {
+            statement.executeUpdate(sqlQuery);
+        }
+    }
+
+    private void showExistingRecordsOfTable(String tableName) throws SQLException {
+        try (Statement statement = connection.createStatement()) {
+            if (tableName != null) {
+                String sqlQuery = String.format("SELECT * FROM \"%s\"", tableName);
+                ResultSet resultSet = statement.executeQuery(sqlQuery);
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: ResultSet is a resource, move it to block "try" (http://bit.ly/2zHWGEE)

File: src/main/java/juja/sqlcmd/Application.java [hunk 3]
```diff
@@ -1,6 +1,153 @@
...
+                int columnCount = resultSet.getMetaData().getColumnCount();
+
+                StringBuilder tableData = new StringBuilder();
+                while (resultSet.next()) {
+                    for (int i = 1; i <= columnCount; i++) {
+                        tableData.append(resultSet.getString(i)).append(" | ");
+                    }
+                    tableData.replace(tableData.length() - 3, tableData.length(), LINE_SEPARATOR);
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: For easier code-reading it's better to extract tableData.length() to separate variable

File: src/main/java/juja/sqlcmd/Application.java [hunk 4]
```diff
@@ -1,6 +1,153 @@
...
+                System.out.println(tableData.toString());
+            }
+        }
+
     }
+
+    private void addUser(String userName, String userPassword) throws SQLException {
+        try (Statement statement = connection.createStatement()) {
+            if (userName != null && userPassword != null) {
+                // Why didn't suit here such expression "INSERT INTO public." + "\"" + TABLE_NAME + "\"" + "(password, name) "
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: I think you just forgot to delete this from here...

File: src/main/java/juja/sqlcmd/Application.java [hunk 5]
```diff
@@ -1,6 +1,153 @@
...
+                //                                      + "VALUES (" + userPassword + ", " + userName + ");";
+                String sqlQuery = String.format("INSERT INTO \"" + TABLE_NAME + "\"(name, password) VALUES('%s','%s') ", userName, userPassword);
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: Why you're using string concatination in String.format? Include TABLE_NAME to expression

File: src/main/java/juja/sqlcmd/Application.java [hunk 6]
```diff
@@ -1,6 +1,153 @@
...
+                statement.executeUpdate(sqlQuery);
+            } else {
+                System.out.println("Wrong input");
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: If you include such behavior in your method then it also should be tested

File: src/main/java/juja/sqlcmd/Application.java [hunk 7]
```diff
@@ -1,6 +1,153 @@
...
+            }
+        }
+    }
+
+    private void dropTableWithName(String user) throws SQLException {
+        try (Statement statement = connection.createStatement()) {
+            String sqlQuery = "DROP TABLE IF EXISTS \"" + user + "\" CASCADE ;";
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: If you start to use String.format - use it everywhere. It's much more usefull than string concatination

File: src/main/java/juja/sqlcmd/Application.java [hunk 8]
```diff
@@ -1,6 +1,153 @@
...
+            statement.executeUpdate(sqlQuery);
+        }
+    }
+
+    private void createTable(String tableName) throws SQLException {
+        try (Statement statement = connection.createStatement()) {
+            String sqlQuery = "CREATE TABLE public.\"" + tableName + "\"" + LINE_SEPARATOR
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: Query doesn't need any LINE_SEPARATORs to be executed. Whole query could be in one line. Gether it with String.format

File: src/main/java/juja/sqlcmd/Application.java [hunk 9]
```diff
@@ -1,6 +1,153 @@
...
+    private void showExistingTablesName() throws SQLException {
+        try (Statement statement = connection.createStatement()) {
+            String sqlQuery = "SELECT" + LINE_SEPARATOR
+                    + "*" + LINE_SEPARATOR
+                    + "FROM" + LINE_SEPARATOR
+                    + "pg_catalog.pg_tables" + LINE_SEPARATOR
+                    + "WHERE" + LINE_SEPARATOR
+                    + "schemaname != 'pg_catalog'" + LINE_SEPARATOR
+                    + "AND schemaname != 'information_schema';";
+            ResultSet rs = statement.executeQuery(sqlQuery);
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java]: ResultSet is a resource, move it to block "try"

File: src/main/java/juja/sqlcmd/Application.java [hunk 10]
```diff
@@ -1,6 +1,153 @@
...
+            } else {
+                System.out.println("db is empty" + LINE_SEPARATOR);
+            }
+        }
+    }
+
+    private void createDbConnection() {
+        try {
+            connection = DriverManager.getConnection(DB_CONNECTION_PATH, DB_USER, DB_PASSWORD);
+        } catch (SQLException e) {
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java:107]: Why you're catching this exception here?

File: src/main/java/juja/sqlcmd/Application.java [hunk 11]
```diff
@@ -1,6 +1,153 @@
...
+            System.out.println("Connection Failed! Check output console");
+            e.printStackTrace();
+        }
+        if (connection != null) {
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java:111]: Pay attention if you'll deside to throw SQLException upper then connection couldn't be null after execution DriverManager.getConnection

File: src/main/java/juja/sqlcmd/Application.java [hunk 12]
```diff
@@ -1,6 +1,153 @@
...
+            System.out.println("Connection created" + LINE_SEPARATOR);
+        } else {
+            System.out.println("Failed to make connection!");
+        }
+    }
+
+    private void checkJDBCDriver() {
```
Comment (Shtramak) [inline src/main/java/juja/sqlcmd/Application.java:118]: I see some mkyong style here))) Logic of this method is a part of createDbConnection(). There's no reason to separate them

File: src/main/java/juja/sqlcmd/Application.java [hunk 13]
```diff
@@ -1,6 +1,153 @@
...
+
+
+    public static void main(String[] arg) throws SQLException {
+        new Application().simpleSQL();
+    }
+
+    public void simpleSQL() throws SQLException {
+        checkJDBCDriver();
+        createDbConnection();
+        dropTableWithName(TABLE_NAME);
```
Comment (adaleorion) [inline src/main/java/juja/sqlcmd/Application.java]: I think logic for prepare tests isn't to be here
We discussed about this case on last meeting
Issue  doesn't have any task about dropping table case
It's test preparation

File: src/main/java/juja/sqlcmd/Application.java [hunk 14]
```diff
@@ -1,6 +1,153 @@
 package juja.sqlcmd;
 
+import java.sql.*;
```
Comment (ivshapovalov) [inline src/main/java/juja/sqlcmd/Application.java]: No * for imports

File: src/main/java/juja/sqlcmd/Application.java [hunk 15]
```diff
@@ -1,6 +1,153 @@
...
+        showExistingTablesName();
+        showExistingRecordsOfTable(TABLE_NAME);
+        updateDataByTitle("user2", "userchange1", TABLE_NAME, "name");
+        showExistingRecordsOfTable(TABLE_NAME);
+        deleteCortegeByTitle(TABLE_NAME, "name", "user3");
+        showExistingRecordsOfTable(TABLE_NAME);
+        connection.close();
+    }
+
+    private void deleteCortegeByTitle(String tableName, String columnName, String title) throws SQLException {
```
Comment (ivshapovalov) [inline src/main/java/juja/sqlcmd/Application.java]: Try to name methods in one pattern. updateData or updateCortege or updateTable etc.

**Review by ivshapovalov — CHANGES_REQUESTED** [review]

File: src/main/java/juja/sqlcmd/Application.java [hunk 16]
```diff
@@ -1,6 +1,153 @@
...
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private Connection connection;
+
+
+    public static void main(String[] arg) throws SQLException {
+        new Application().simpleSQL();
+    }
+
+    public void simpleSQL() throws SQLException {
+        checkJDBCDriver();
```
Comment (kortov) [inline src/main/java/juja/sqlcmd/Application.java:27]: It's better to use Capital letters for 2 long acronyms and Camel case for 3 or long https://goo.gl/CgvmeP

File: src/main/java/juja/sqlcmd/Application.java [hunk 17]
```diff
@@ -1,6 +1,153 @@
...
+        createDbConnection();
+        dropTableWithName(TABLE_NAME);
+        showExistingTablesName();
+        createTable(TABLE_NAME);
+        for (int i = 1; i < 4; i++) {
+            addUser("user" + i, "password" + i);
+        }
+        showExistingTablesName();
+        showExistingRecordsOfTable(TABLE_NAME);
+        updateDataByTitle("user2", "userchange1", TABLE_NAME, "name");
```
Comment (kortov) [inline src/main/java/juja/sqlcmd/Application.java]: Why do you pass in as a parameter tableName to update method but don't do it with addUser? I think 4 parameters for method is pretty much. https://goo.gl/7cVP2j

File: src/main/java/juja/sqlcmd/Application.java (hunk 14 above)
Comment (kortov) [reply src/main/java/juja/sqlcmd/Application.java]: + 1 java code convention

File: src/main/java/juja/sqlcmd/Application.java (hunk 13 above)
Comment (kortov) [reply src/main/java/juja/sqlcmd/Application.java]: Anyway we all did it)

File: src/main/java/juja/sqlcmd/Application.java (hunk 15 above)
Comment (kortov) [reply src/main/java/juja/sqlcmd/Application.java]: +1

File: src/main/java/juja/sqlcmd/Application.java (hunk 2 above)
Comment (kortov) [reply src/main/java/juja/sqlcmd/Application.java]: +1

**Review by kortov — COMMENTED** [review]
Pretty good, just make little improvements. And better clean output, I think there are many redundant lines

File: src/main/java/juja/sqlcmd/Application.java (hunk 15 above)
Comment (DenysMelnychenko) [reply src/main/java/juja/sqlcmd/Application.java]: Thank's for your coment. Will change to one pattern in next commit.

**Comment by adaleorion** [comment]
Hi, @DenysMelnychenko 
How are you? Do you check comments in this task?
Can you make some changes?

**Comment by DenysMelnychenko** [comment]
HI. I feel better now. Yes ofcouse i will make changes. Trully, i already done them, But i can't add it to existing pull request.

File: pom.xml [hunk 18]
```diff
@@ -2,16 +2,30 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>juja.sqlcmd</groupId>
+    <groupId>juja</groupId>
     <artifactId>sqlcmd</artifactId>
     <version>1.0-SNAPSHOT</version>
+    <build>
```
Comment (kortov) [inline pom.xml]: both blocks do the same functionality but java versions inside of them are different

File: pom.xml [hunk 19]
```diff
@@ -2,16 +2,30 @@
...
+        </plugins>
+    </build>
     <packaging>jar</packaging>
 
     <name>sqlcmd</name>
     <url>http://maven.apache.org</url>
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (kortov) [inline pom.xml:27]: both blocks do the same functionality but java versions inside of them are different

File: src/main/java/juja/sqlcmd/Application.java [hunk 20]
```diff
@@ -1,6 +1,131 @@
...
+
 public class Application {
-    public static void main(String[] args) {
+
+    private static final String POSTGRESQL_DRIVER = "org.postgresql.Driver";
+    private static final String DB_CONNECTION_PATH = "jdbc:postgresql://127.0.0.1:5432/sqlcmd";
+    private static final String DB_USER = "sqlcmd";
+    private static final String DB_PASSWORD = "sqlcmd";
+    private static final String TABLE_NAME = "user";
+    public static final String SHOW_EXISTING_TABLES_NAME = "SELECT * FROM pg_catalog.pg_tables WHERE schemaname != 'pg_catalog' AND schemaname != 'information_schema';";
```
Comment (kortov) [inline src/main/java/juja/sqlcmd/Application.java:17]: You can implement it without inversions as "SELECT table_name FROM information_schema.tables WHERE table_schema='public' AND table_type='BASE TABLE'" .
And why is it constant?

**Review by kortov — COMMENTED** [review]
It looks quite good

File: pom.xml (hunk 19 above)
Comment (DenysMelnychenko) [reply pom.xml:27]: Yes, you are right. Will fix it. ;)

File: src/main/java/juja/sqlcmd/Application.java (hunk 20 above)
Comment (DenysMelnychenko) [reply src/main/java/juja/sqlcmd/Application.java:17]: It is constant because we will never change the order or schema name for this functionality.

**Review by ivshapovalov — APPROVED** [review]

**Review by Shtramak — APPROVED** [review]

**Comment by adaleorion** [comment]
Thanks, it good, but in the future please use names for your commits by this format `#number-task text`

**Review by adaleorion — APPROVED** [review]

**Pull request merged**